pkg/app/piped/analysisprovider/log/provider.go

```diff
@@ -22,6 +22,7 @@ import (
 type Provider interface {
 	Type() string
 	// RunQuery runs the given query against the log provider,
-	// and then checks if there is at least one error log..
-	RunQuery(ctx context.Context, query string) (result bool, err error)
+	// and then checks if there is at least one error log.
+	// Returns the result reason if non-error occurred.
+	RunQuery(ctx context.Context, query string) (result bool, reason string, err error)
```

Comment:
Looks like now it should be `Evaluate` instead of `RunQuery`.
But that is not in this PR scope.

Reply:
@nghialv Right. Let me address it later.